src/gettext.js

```diff
@@ -42,9 +42,14 @@ function gettext(string, replacements, locale_parameter) {
     }
     var locale = locale_parameter || locale_default;
 
-    if (locale && catalogs[locale] && catalogs[locale].messages && catalogs[locale].messages[string] &&
-        catalogs[locale].messages[string].length > 0 && catalogs[locale].messages[string][0] !== "") {
-        result = catalogs[locale].messages[string][0];
+    if (locale && catalogs[locale] && catalogs[locale].messages && catalogs[locale].messages[string]) {
+        var messages = catalogs[locale].messages[string];
+        if (messages instanceof Array && messages.length > 0 && messages[0] !== "") {
```

Comment:
The use of `Array.isArray(messages)` is more robust

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#instanceof_vs_array.isarray